File: src/main/java/de/akquinet/ccsp/floodwatch/datamodel/Location.java
package de.akquinet.ccsp.floodwatch.datamodel;

import lombok.Data;
import org.springframework.data.mongodb.core.geo.GeoJsonPoint;
import org.springframework.data.mongodb.core.mapping.Document;

@Document
@Data
public class Location {

    private String id;

    private final String externalId;

    private final GeoJsonPoint geoLocation;

    private float altitude;

    private final String waterBody;

    private final String surveyStation;

    private final String displayName;
}

File: src/main/kotlin/de/akquinet/ccsp/floodwatch/datamodel/Location.kt
package de.akquinet.ccsp.floodwatch.datamodel

import org.springframework.data.mongodb.core.geo.GeoJsonPoint
import org.springframework.data.mongodb.core.mapping.Document

@Document
data class Location(val externalId: String,
                    val geoLocation: GeoJsonPoint,
                    val waterBody: String,
                    val surveyStation: String,
                    val displayName: String) {
    var id: String? = null
    var altitude: Float? = null
}
